vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,187 @@
+			saveQueryResults.printErrorAndExit();
+		}
+
+		// TODO
+		// if (!saveModel.isConfigOk()) {
+		// saveModel.printErrorAndExit();
+		// }
+
+		System.out.println("Configuration:");
+
+		/* Configure rules */
+		final KnowledgeBase kb = new KnowledgeBase();
+		for (String ruleFile : ruleFiles) {
+			try {
+				RuleParser.parseInto(kb, new FileInputStream(ruleFile));
+				System.out.println("  --rule-file: " + ruleFile);
+			} catch (FileNotFoundException e) {
+				System.out.println("File not found: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse rule file: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Configure queries (We throw parsing query errors before materialization) */
+		List<PositiveLiteral> queries = new ArrayList<>();
+		for (String queryString : queryStrings) {
+			try {
+				queries.add(RuleParser.parsePositiveLiteral(queryString));
+				System.out.println("  --query: " + queries.get(queries.size() - 1));
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse query: " + queryString + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Print configuration */
+		System.out.println("  --log-file: " + logFile);
+		System.out.println("  --log-level: " + logLevel);
+		System.out.println("  --chase-algorithm: " + chaseAlgorithm);
+		System.out.println("  --timeout: " + ((timeout > 0) ? Integer.toString(timeout) : "none"));
+
+		/* Print what to do with the result */
+
+		printQueryResults.printConfiguration();
+		saveQueryResults.printConfiguration();
+		// saveModel.printConfiguration();
+
+		try (Reasoner reasoner = new VLogReasoner(kb)) {
+			// logFile
+			reasoner.setLogFile(logFile);
+			// logLevel
+			reasoner.setLogLevel(logLevel);
+			// chaseAlgorithm
+			reasoner.setAlgorithm(chaseAlgorithm);
+			// timeout
+			if (timeout > 0) {
+				reasoner.setReasoningTimeout(timeout);
+			}
+
+			try {
+				System.out.println("Executing the chase ...");
+				reasoner.reason();
+			} catch (IOException e) {
+				System.out.println("Something went wrong. Please check the log file.");
+				System.exit(1);
+			}
+
+			// TODO save the model
+			// if (saveModel.saveModel) {
+			// System.out.println("Saving model ...");
+			// }
+
+			String outputPath;
+			if (queries.size() > 0) {
+				System.out.println("Answering queries ...");
+				for (PositiveLiteral query : queries) {
+					if (saveQueryResults.saveResults) {
+						try {
+							outputPath = saveQueryResults.outputQueryResultDirectory + "/" + query + ".csv";
+							reasoner.exportQueryAnswersToCsv(query, outputPath, true);
+						} catch (IOException e) {
+							System.out.println("Can't save query " + query);
+							System.exit(1);
+						}
+					}
+					if (printQueryResults.sizeOnly) {
+						System.out.println(
+								"Elements in " + query + ": " + ExamplesUtils.getQueryAnswerCount(query, reasoner));
```

Comment:
Number of query answers.

Why call query answers "elements"?

Reply:
Thank you